includes/gateway/class-omise-payment-fbbot.php

```diff
@@ -0,0 +1,266 @@
+
+		    $order = Omise_FBBot_WooCommerce::create_order( $params );
+
+		    $metadata = array(
+				'source' => 'woo_omise_bot',
+				'product_id' => $product_id,
+				'messenger_id' => $messenger_id,
+				'order_id' => $order->get_order_number()
+		    );
+
+		    $data = array(
+				'amount'      => $this->format_amount_subunit( $order->get_total(), $order->get_currency() ),
+				'currency'    => $order->get_currency(),
+				'description' => 'OrderID is '.$order->get_id().' : This order created from Omise FBBot and CustomerID is '.$messenger_id,
+				'metadata' => $metadata,
+				'card' => $omise_token
+		    );
+
+		    if ( $this->omise_3ds ) {
+		    	$return_uri =  site_url() . '/complete-payment';
+
+		      	$data['return_uri'] = $return_uri;
+		    }
+
+		    // Create Charge
+		    try {
+		      	$charge = OmiseCharge::create( $data, '', $this->secret_key() );
+		      	// We move checking charge status to request handler in handle triggered from omise method
+
+		      	// Just sent message to user for let them know we received these order
+		      	$prepare_confirm_message = Omise_FBBot_Conversation_Generator::prepare_confirm_order_message( $order->get_order_number() );
+		      	$response = Omise_FBBot_HTTPService::send_message_to( $messenger_id, $prepare_confirm_message );
+
+		      	// If merchant enable 3ds mode
+		      	if ( isset ( $charge['authorize_uri'] ) ) {
+		        	if ( wp_redirect( $charge['authorize_uri'] ) ) {
+		          		error_log( 'redirect to -> '. $charge['authorize_uri'] );
+		          		exit;
+		        	}
+		        
+		        	return;
+		      	}
+
+		      	// If merchant disable 3ds mode : normal mode
+		      	$redirect_uri =  site_url() . '/complete-payment';
+		      	if ( wp_redirect( $redirect_uri ) ) {
+		          	exit;
+		      	}
+
+		    } catch (Exception $e) {
+		      	error_log("catch error : " . $e->getMessage());
+
+		      	$error_message = str_replace(" ", "%20", $e->getMessage());
+
+		      	// [WIP] - Redirect to error page
+		      	$redirect_uri =  site_url() . '/pay-on-messenger-error/?error_message=' . $error_message;
+		      	if ( wp_redirect( $redirect_uri ) ) {
+		          	exit;
+		      	}
+		    }
+  		}
+
+  		private function is_omise_payment_page() {
+			global $wp;
+
+			if ( strtolower( $wp->request ) == $this->payment_page_url ) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private function is_omise_payment_error_page() {
+			global $wp;
+			
+			if ( strtolower( $wp->request ) == $this->payment_error_url ) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private function is_omise_purchase_complete_page() {
+			global $wp;
+			
+			if ( strtolower( $wp->request ) == $this->payment_purchase_complete_url ) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private function is_accessible() {
+			global $wp;
+
+			if ( strtolower( $wp->request ) == $this->payment_page_url || strtolower( $wp->request ) == $this->payment_error_url ) {
```

Comment:
Yes, Like that you advice me for about the return case in PHP style. Make it short and clear 👍  🙇